hera_mc/cm_redis_corr.py

```diff
@@ -0,0 +1,158 @@
+        Pass through of redis session
+
+    Returns
+    -------
+    snap_to_ant : dict
+        Dictionary mapping the snaps to the antennas
+    ant_to_snap : dict
+        Dictionary mapping each antenna to its snap
+    """
+    snap_to_ant = {}
+    ant_to_snap = {}
+    for antn, ant in enumerate(cminfo['antenna_numbers']):
+        name = cminfo['antenna_names'][antn]
+        e_pol = None
+        n_pol = None
+        for pol in cminfo['correlator_inputs'][antn]:
+            if pol.lower().startswith('e') and '>' in pol:
+                e_pol = pol
+            if pol.lower().startswith('n') and '>' in pol:
+                n_pol = pol
+        ant_to_snap[ant] = {}
+        if e_pol is not None:
+            snapi_e, channel_e = snap_part_to_host_input(e_pol, rsession=rsession)
+            ant_to_snap[ant]['e'] = {'host': snapi_e, 'channel': channel_e}
+            snap_to_ant.setdefault(snapi_e, [None] * 6)
+            snap_to_ant[snapi_e][channel_e] = name + 'E'
+        if n_pol is not None:
+            snapi_n, channel_n = snap_part_to_host_input(n_pol, rsession=rsession)
+            ant_to_snap[ant]['n'] = {'host': snapi_n, 'channel': channel_n}
+            snap_to_ant.setdefault(snapi_n, [None] * 6)
+            snap_to_ant[snapi_n][channel_n] = name + 'N'
+    return snap_to_ant, ant_to_snap
+
+
+def cminfo_redis_loc(cminfo):
+    """
+    Places the positional data into redis for use by the correlator.
+
+    Parameter
+    ---------
+    cminfo : dict
+        Dictionary as returned from get_cminfo_correlator()
+
+    Returns
+    -------
+    ant_pos : dict
+        Dictionary containing the ECEF positions of all of the antennas.
+    cofa : dict
+        Dictionary containing the lat, lon, alt of the center-of-array
+    """
+    locations = {}
+    locations['antenna_numbers'] = cminfo['antenna_numbers']
+    locations['antenna_names'] = cminfo['antenna_names']
+    antenna_positions = []
+    for _x in cminfo['antenna_positions']:
+        antenna_positions.append(list(_x))
+    locations['antenna_positions'] = antenna_positions
+    locations['antenna_utm_eastings'] = cminfo['antenna_utm_eastings']
+    locations['antenna_utm_northings'] = cminfo['antenna_utm_northings']
+    locations['cofa'] = {'lat': cminfo['cofa_lat'], 'lon': cminfo['cofa_lon'],
+                         'alt': cminfo['cofa_alt']}
+    return locations
+
+
+def set_redis_cminfo(redishost=DEFAULT_REDIS_ADDRESS, session=None):
+    """
+    Gets the configuration management information and writes to the
+    redis database for the correlator.
+
+    Parameters
+    ----------
+    redishost : None or str
+        Hostname for the redis database.  If None uses default
+    session : None or hera_mc session
+        Session for hera_mc instance.  None uses default
+    """
+    h = cm_sysutils.Handling(session=session)
+    cminfo = h.get_cminfo_correlator()
+
+    if redishost is None:
+        redishost = DEFAULT_REDIS_ADDRESS
```

Comment:
Now that you aren't defaulting it to None, I don't think you need these lines.

Reply:
I believe that @mkolopanis wanted it to go to the default if you explicitly fed it None.  I don't think I have a preference either way.

Reply:
Ok, he understands this better than I do, but I would have thought None would just error, which would be fine.

Reply:
I think this `if` statement is only necessary if we leave the input redishost as `None`. Dave and I talked before privately about doing something like this in some other code I wrote, but if @bhazelton  wants the default to be `DEFAULT_REDIS_ADDRESS` then we should just delete this `if` statement.

Reply:
This matches the way it is done in autocorrelations, so we should change both if that is what we want.

Reply:
Now that @mkolopanis has verified that using `None` doesn't error but instead uses `redishost` I'm ok with leaving them.

Reply:
Maybe add a comment explaining that so people know why it's there.